services/streammanager/googlepubsub/googlepubsubmanager.go

```diff
@@ -0,0 +1,198 @@
+	if err != nil {
+		return nil, fmt.Errorf("[GooglePubSub] Error while marshalling destination config :: %w", err)
+	}
+	err = json.Unmarshal(jsonConfig, &config)
+	if err != nil {
+		return nil, fmt.Errorf("[GooglePubSub] error  :: error in GooglePubSub while unmarshelling destination config:: %w", err)
+	}
+	var client *pubsub.Client
+	if config.Credentials != "" && config.ProjectId != "" {
+		client, err = pubsub.NewClient(ctx, config.ProjectId, option.WithCredentialsJSON([]byte(config.Credentials)))
+	}
+	if err != nil {
+		return nil, err
+	}
+	var eventToTopic = make([]map[string]interface{}, len(config.EventToTopicMap))
+	for i, s := range config.EventToTopicMap {
+		topic := client.Topic(s["to"])
+		if eventToTopic[i] == nil {
+			eventToTopic[i] = make(map[string]interface{})
+		}
+		eventToTopic[i]["event"] = s["from"]
+		eventToTopic[i]["topic"] = topic
+	}
+	var pbsClient *pubsubClient
+	pbsClient = &pubsubClient{client, eventToTopic}
+	return pbsClient, nil
+}
+func Produce(jsonData json.RawMessage, producer interface{}, destConfig interface{}) (statusCode int, respStatus string, responseMessage string) {
+	parsedJSON := gjson.ParseBytes(jsonData)
+	pbs, ok := producer.(*pubsubClient)
+	ctx := context.Background()
+	if !ok {
+		respStatus = "Failure"
+		responseMessage = "[GooglePubSub] error :: Could not create producer"
+		return 400, respStatus, responseMessage
+	}
+	var data interface{}
+	if parsedJSON.Get("message").Value() != nil {
+		data = parsedJSON.Get("message").Value()
+	} else {
+		respStatus = "Failure"
+		responseMessage = "[GooglePubSub] error :: message from payload not found"
+		return 400, respStatus, responseMessage
+	}
+	value, err := json.Marshal(data)
+
+	if err != nil {
+		respStatus = "Failure"
+		responseMessage = "[GooglePubSub] error  :: " + err.Error()
+		logger.Errorf("[GooglePubSub] error  :: %w", err)
+		statusCode := 400
+		return statusCode, respStatus, responseMessage
+	}
+	if parsedJSON.Get("topicId").Value() != nil {
+		topicIdString, ok := parsedJSON.Get("topicId").Value().(string)
+		if !ok {
+			respStatus = "Failure"
+			responseMessage = "[GooglePubSub] error :: Could not parse topic id to string"
+			logger.Error(responseMessage)
+			statusCode := 400
+			return statusCode, respStatus, responseMessage
+		}
+		if topicIdString == "" {
+			respStatus = "Failure"
+			responseMessage = "[GooglePubSub] error :: empty topic id string"
+			return 400, respStatus, responseMessage
+		}
+		var topic *pubsub.Topic
+		for _, s := range pbs.EventToTopic {
+			t := s["topic"].(*pubsub.Topic)
+			splitString := strings.Split(t.String(), "/")
+			if splitString[3] == topicIdString {
+				topic = t
+				break
+			}
+		}
+		if topic == nil {
+			statusCode = 400
+			responseMessage = "[GooglePubSub] error :: Topic not found in project"
+			respStatus = "Failure"
+			return statusCode, respStatus, responseMessage
+		}
+		topic.PublishSettings.DelayThreshold = 0
```

Comment:
Setting this once should be enough, while topic creation.

Reply:
Are we doing load tests? Did we observe any difference because of this change?

Reply:
done.
Will do load test.